### PR TITLE
Resolve issue #48

### PR DESCRIPTION
**Pull request by vdakalov**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -170,7 +170,7 @@ test('removing regex pattern without other keys is supported', function (t) {
   t.plan(2)
 
   var instance = bloomrun()
-  var pattern = { to: /.*/ }
+  var pattern = { to: /^[a-zA-Z0-9-.]+$/i }
```
Comment (mcollina) [inline test.js]: Can you add a new test instead?

**Comment by mcollina** [comment]
cc @StarpTech

File: test.js (hunk 1 above)
Comment (vdakalov) [reply test.js]: only if you will suggest description of it

Comment (mcollina) [reply test.js]: name the one with the wildcard ' - wildcard case'.

File: lib/match.js [hunk 2]
```diff
@@ -3,14 +3,15 @@
...
 
 function equal (a, b) {
-  var result = false
   if (typeof a !== 'object') {
-    result = safeEqual(a, b)
+    return safeEqual(a, b)
   } else if (a instanceof RegExp) {
-    result = a.test(b)
+    if (b instanceof RegExp) {
+      return b.toString() === b.toString()
```
Comment (StarpTech) [inline lib/match.js]: I think comparing `regex.source` is better.

Comment (vdakalov) [reply lib/match.js]: agree

File: test.js [hunk 3]
```diff
@@ -181,6 +181,21 @@ test('removing regex pattern without other keys is supported', function (t) {
   t.equal(instance.lookup({ to: 'you' }), null)
 })
 
+test('removing regex pattern without other keys is supported - wildcard case', function (t) {
+  t.plan(2)
+
+  var instance = bloomrun()
+  var pattern = { to: /^[a-zA-Z0-9-.]+$/i }
+
+  instance.add(pattern)
```
Comment (StarpTech) [inline test.js]: Please add a negative case here to ensure that it doesn't remove all patterns at once.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: test.js (hunk 3 above)
Comment (vdakalov) [reply test.js]: No. I've already spent a lot of time on your code. I don't have any more time. If you know how to make these edits better - do it! I'm done!

**Review by StarpTech — APPROVED** [review]
LGTM after my changes

**Review by mcollina — APPROVED** [review]
LGTM

**Pull request merged**